packages/@eventual/aws-runtime/src/stores/bucket-store.ts

```diff
@@ -0,0 +1,175 @@
+  ): Promise<PutBucketObjectResponse> {
+    const result = await this.props.s3.send(
+      new PutObjectCommand({
+        Bucket: this.bucketName(bucketName),
+        Key: key,
+        Body: data,
+      })
+    );
+
+    return {
+      etag: result.ETag,
+    };
+  }
+
+  async delete(bucketName: string, key: string): Promise<void> {
+    await this.props.s3.send(
+      new DeleteObjectCommand({
+        Bucket: this.bucketName(bucketName),
+        Key: key,
+      })
+    );
+  }
+
+  async copyTo(
+    bucketName: string,
+    key: string,
+    destKey: string,
+    destBucket?: Bucket | undefined,
+    options?: CopyBucketObjectOptions | undefined
+  ): Promise<CopyBucketObjectResponse> {
+    const result = await this.props.s3.send(
+      new CopyObjectCommand({
+        Bucket: this.bucketName(bucketName),
+        Key: destKey,
+        CopySource: `${
+          destBucket
+            ? this.bucketName(destBucket.name)
+            : this.bucketName(bucketName)
+        }/${key}`,
+        CopySourceIfMatch: options?.sourceEtag,
+      })
+    );
+
+    return {
+      etag: result.CopyObjectResult?.ETag,
+    };
+  }
+
+  async presignedUrl(
+    bucketName: string,
+    key: string,
+    operation: "get" | "put" | "delete",
+    expires?: DurationSchedule | undefined
+  ): Promise<{ url: string; expires: string }> {
+    const request =
+      operation === "get"
+        ? new GetObjectCommand({
+            Bucket: this.bucketName(bucketName),
+            Key: key,
+          })
+        : operation === "put"
+        ? new PutObjectCommand({
+            Bucket: this.bucketName(bucketName),
+            Key: key,
+          })
+        : operation === "delete"
+        ? new DeleteObjectCommand({
+            Bucket: this.bucketName(bucketName),
+            Key: key,
+          })
+        : assertNever(operation);
+    const _expires = expires ? computeDurationSeconds(expires) : 3600;
+    const url = await getSignedUrl(this.props.s3, request, {
+      expiresIn: _expires,
+    });
+
+    return {
+      expires: new Date(new Date().getTime() + _expires * 1000).toISOString(),
+      url,
+    };
+  }
+
+  async list(
+    bucketName: string,
+    request: ListBucketRequest
+  ): Promise<ListBucketResult> {
+    const result = await this.props.s3.send(
+      new ListObjectsV2Command({
+        Bucket: this.bucketName(bucketName),
+        Prefix: request.prefix,
+        ContinuationToken: request.nextToken,
+        StartAfter: request.startAfter,
+        MaxKeys: request.maxKeys,
+      })
+    );
+
+    return {
+      keyCount: result.KeyCount ?? 0,
+      nextToken: result.NextContinuationToken,
+      objects:
+        result.Contents?.filter((c) => c.Key).map((c) => ({
+          eTag: c.ETag!,
+          key: c.Key!,
+          size: c.Size ?? 0,
+        })) ?? [],
+    };
+  }
+
+  private bucketName(bucketName: string) {
+    return bucketServiceBucketName(getLazy(this.props.serviceName), bucketName);
+  }
```

Comment:
Should this be a public getter?

Reply:
It's just a local helper, any use of bucket name is internal.

Reply:
That makes escape hatches to aws s3 sdk impossible. Why not expose it?

Reply:
We don't expose the raw resource name for anything else. Won't it be confusing for a Bucket to have a name and a bucket name? 

We could create a property called `awsBucketName`. I assume the bucket name discussed above would be defined in the service side? 

If the override is defined in the CDK side, we'd need to rely on synthesis and maybe deployment to get the value (not the service spec like all other values). This value would not exist for local/testing as well as local doesn't have AWS. the `awsBucketName` would be undefned.

I'd argue lets follow the current pattern of hiding the aws layer until this value is needed.

Reply:
I don't think it's confusing. It's safe to assume a bucket in any system has a physical name.

Reply:
Yeah, thats true, and the physical name can be different between envs.